sdk/eventhub/azure-eventhub/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,223 @@
+
+### Migrate from legacy to new client library
+
+The [migration guide][MigrationGuide] includes steps on migrating from the legacy client and migrating legacy checkpoints.
+
+## Get additional help
+
+Additional information on ways to reach out for support can be found in the [SUPPORT.md][SUPPORT] at the repo's root.
+
+### Filing GitHub issues
+
+When filing GitHub issues, the following details are requested:
+
+* Event Hub environment
+  * How many partitions?
+* EventHubConsumerClient environment
+  * What is the machine(s) specs processing your Event Hub?
+  * How many instances are running?
+* What is the average size of each EventData?
+* What is the traffic pattern like in your Event Hub?  (i.e. # messages/minute and if the EventHubConsumerClient is always busy or has slow traffic periods.)
+* Repro code and steps
+  * This is important as we often cannot reproduce the issue in our environment.
+* Logs.  We need DEBUG logs, but if that is not possible, INFO at least.  Error and warning level logs do not provide enough information.  The period of at least +/- 10 minutes from when the issue occurred.
+
+<!-- repo links -->
+[IoTConnectionString]: https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/eventhub/azure-eventhub/samples/async_samples/iot_hub_connection_string_receive_async.py
+
+<!-- docs.microsoft.com links -->
+[ExceptionModule]: https://docs.microsoft.com/python/api/azure-eventhub/azure.eventhub.exceptions
+[EventHubError]: https://docs.microsoft.com/python/api/azure-eventhub/azure.eventhub.exceptions.eventhuberror
+[AmqpTroubleshooting]: https://docs.microsoft.com/azure/service-bus-messaging/service-bus-amqp-troubleshoot
+[AuthorizeSAS]: https://docs.microsoft.com/azure/event-hubs/authorize-access-shared-access-signature
+[RBACRoles]: https://docs.microsoft.com/azure/event-hubs/troubleshoot-authentication-authorization
+[Epoch]: https://docs.microsoft.com/azure/event-hubs/event-hubs-event-processor-host#epoch
+[EventHubsIPAddresses]: https://docs.microsoft.com/azure/event-hubs/troubleshooting-guide#what-ip-addresses-do-i-need-to-allow
+[EventHubsMessagingExceptions]: https://docs.microsoft.com/azure/event-hubs/event-hubs-messaging-exceptions
+[EventHubsTroubleshooting]: https://docs.microsoft.com/azure/event-hubs/troubleshooting-guide
+[GetConnectionString]: https://docs.microsoft.com/azure/event-hubs/event-hubs-get-connection-string
+[IoTEventHubEndpoint]: https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-messages-read-builtin
+[IoTHubSAS]: https://docs.microsoft.com/azure/iot-hub/iot-hub-dev-guide-sas#security-tokens
+[troubleshoot_authentication_authorization]: https://docs.microsoft.com/azure/event-hubs/troubleshoot-authentication-authorization
```

Comment:
nit:  The format of this differs from the others.   Should we consider making this PascalCase or changing the others to snake_case?

Reply:
Ill make them all PascalCase, thats how the majority of them are set up